examples/recording_service/leveldb-integration/cpp/README.md

```diff
@@ -0,0 +1,1077 @@
+against the cross-compiled Connext binaries. To fix this, you have to create a
+file with the architecture name and call CMake with a specific flag called
+``-DCMAKE_TOOLCHAIN_FILE``. An example of the file to create with the toolchain
+settings (e.g. for an ARM architecture):
+
+```cmake
+set(CMAKE_SYSTEM_NAME Linux)
+set(toolchain_path "<path to>/arm-bcm2708/gcc-linaro-arm-linux-gnueabihf-raspbian")
+set(CMAKE_C_COMPILER "${toolchain_path}/bin/arm-linux-gnueabihf-gcc")
+set(CMAKE_CXX_COMPILER "${toolchain_path}/bin/arm-linux-gnueabihf-g++")
+```
+
+Then you can call CMake like this:
+
+```bash
+cmake -DCONNEXTDDS_DIR=<connext dir> 
+      -DCMAKE_TOOLCHAIN_FILE=<toolchain file created above>
+      -DCONNEXTDDS_ARCH=<connext architecture>
+      -DCMAKE_BUILD_TYPE=Release 
+      -DLEVELDB_DIR=<leveldb directory> 
+      -DLEVELDB_INCLUDE=<leveldb include directory> 
+      ..
+```
+
+## Running *Recorder*
+
+First of all, you need to make sure that the `LevelDbWriter` shared library
+is in the system's library path.  On Mac OS X, you must use the variable
+`RTI_LD_LIBRARY_PATH` instead of the common `DYLD_LIBRARY_PATH`.
+
+This example will work with any DDS Topic discovered on the DDS Domain ID of 
+your choice. Run any application publishing data into that domain, for example
+*RTI Shapes Demo* or *RTI DDS Ping*.
+
+The supplied XML configuration for *Recorder* expects you to set an environment 
+variable called `LEVELDB_WORKING_DIR`. This variable should point to a directory 
+with write permissions and where all the LevelDB directories and files will be 
+created.
+The path provided should not contain any trailing path separators. This variable
+is used in the XML file to set the XML variable called 
+`rti.recording.examples.leveldb.working_dir`.
+
+> **Note**:
+> Note: in this example, *Recorder* sets the `error_if_exists` flag when
+> opening the database files (directories). This will make the application fail
+> to run if the database was already recorded. To comfortably avoid having to
+> set the `LEVELDB_WORKING_DIR` variable to a different value, or having to
+> remove the contents of the directory each time we want to run *Recorder*
+> consecutively, an XML variable is provided: 
+> `rti.recording.examples.leveldb.enable_auto_dir`. When enabled, *Recorder*
+> will append the current timestamp in seconds to the working directory 
+> specified by the `LEVELDB_WORKING_DIR` environment variable and/or the 
+> `rti.recording.examples.leveldb.working_dir` XML variable.
+
+To run the example, you need to run the following command from the `build`
+folder (where the storage writer plugin shared library has been created).
+
+```bash
+export LEVELDB_WORKING_DIR=<storage location>
+cd build
+<connext dir>/bin/rtirecordingservice 
+        -cfgFile ../leveldb_recorder.xml
+        -cfgName LevelDbIntegration 
+        -domainIdBase <your chosen domain ID>
+```
+
+Note that if you don't want to export the necessary environment variables
```

Comment:
I would switch this. Promote the -D as the way to do it, and do a note indicating that the enviornment can be used if preferred.
I find setting environment rather annoying, so I'd like to promote more an easier way to use variables

Reply:
I've switched the order and the text so that `-D` is promoted